cmd/snap-seccomp/main_test.go

```diff
@@ -217,26 +232,58 @@ set_tls
 # arm64
 readlinkat
 faccessat
+# i386 from amd64
+restart_syscall
 `
 	bpfPath := filepath.Join(c.MkDir(), "bpf")
 	err = main.Compile([]byte(common+seccompWhitelist), bpfPath)
 	c.Assert(err, IsNil)
 
+	// default syscall runner
+	syscallRunner := s.seccompSyscallRunner
+
 	// syscallName;arch;arg1,arg2...
 	l := strings.Split(bpfInput, ";")
-	if len(l) > 1 && l[1] != "native" {
-		c.Logf("cannot use non-native in runBpfInKernel")
-		return
+	syscallName := l[0]
+	syscallArch := "native"
+	if len(l) > 1 {
+		syscallArch = l[1]
 	}
 
-	var syscallRunnerArgs [7]string
-	syscallNr, err := seccomp.GetSyscallFromName(l[0])
+	syscallNr, err := seccomp.GetSyscallFromName(syscallName)
 	c.Assert(err, IsNil)
+
+	// Check if we want to test non-native architecture
+	// handling. Doing this via the in-kernel tests is tricky as
+	// we need a kernel that can run the architecture and a
+	// compiler that can produce the required binaries. Currently
+	// we only test amd64 running i386 here.
+	if syscallArch != "native" {
+		syscallNr, err = seccomp.GetSyscallFromNameByArch(syscallName, main.UbuntuArchToScmpArch(syscallArch))
+		c.Assert(err, IsNil)
+
+		switch syscallArch {
+		case "amd64":
+			// default syscallRunner
+		case "i386":
+			syscallRunner = s.seccompSyscallRunner + ".m32"
+		default:
+			c.Errorf("unsupported non-native arch: %s", syscallArch)
```

Comment:
This doesn't seem quite right. While it is fine to only test i386 compat on amd64, non-native armhf with arm64 is not 'unsupported' per se, it is just not valid for this particular test.

Perhaps:
```
c.Errorf("unexpected non-native syscall for arch: %s", syscallArch)
```